tests/test_mean_functions.py

```diff
@@ -40,3 +51,37 @@ def test_constant(constant: Float[Array, " Q"]) -> None:
     assert (
         mf(jnp.array([[1.0, 2.0], [3.0, 4.0]])) == jnp.array([constant, constant])
     ).all()
+
+
+def test_zero_mean_remains_zero() -> None:
+    key = jr.PRNGKey(123)
+
+    x = jr.uniform(key=key, minval=0, maxval=1, shape=(20, 1))
+    y = jnp.full((20, 1), 50, dtype=jnp.float64)  # Dataset with non-zero mean
+    D = gpx.Dataset(X=x, y=y)
+
+    kernel = gpx.kernels.Constant(constant=jnp.array(0.0))
+    kernel = kernel.replace_trainable(
+        constant=False
+    )  # Prevent kernel from modelling non-zero mean
+    meanf = Zero()
+    prior = gpx.Prior(mean_function=meanf, kernel=kernel)
+    likelihood = gpx.Gaussian(num_datapoints=D.n, obs_noise=jnp.array(1e-6))
+    likelihood = likelihood.replace_trainable(obs_noise=False)
```

Comment:
And trainability turned off here too?

Reply:
I just turned off trainability in both cases such that if the `Zero` mean was still trainable, it would be the only part of the model able to be trained (and so no other aspect of the model (such as the kernel) could be used to model the non-zero mean of the data other than the `Zero` mean function).